simvue/factory/dispatch/queued.py

```diff
@@ -59,26 +61,31 @@ def __init__(
         max_read_rate : float
             maximum rate at which the callback can be executed
         """
+        DispatcherBaseClass.__init__(
+            self,
+            callback=callback,
+            object_types=object_types,
+            termination_trigger=termination_trigger,
+        )
         super().__init__()
 
         self._termination_trigger = termination_trigger
         self._callback = callback
-        self._queues = {label: queue.Queue() for label in queue_categories}
+        self._queues = {label: queue.Queue() for label in object_types}
         self._max_read_rate = max_read_rate
         self._max_buffer_size = max_buffer_size
         self._send_timer = 0
-        self._queue_blocking = queue_blocking  # Does nothing
 
-    def add_item(self, item: typing.Any, queue_label: str, blocking: bool) -> None:
+    def add_item(self, item: typing.Any, object_type: str, blocking: bool) -> None:
```

Comment:
The required "blocking" parameter gives this a different interface to the direct dispatcher, which is not ideal. I think it should be at least defaulted.